core_impl/src/test/java/com/google/instrumentation/stats/StatsManagerImplTest.java

```diff
@@ -14,243 +14,457 @@
+      MeasurementDescriptor.Name.create("my measurement");
+
+  private static final MeasurementDescriptor.Name MEASUREMENT_NAME2 =
+      MeasurementDescriptor.Name.create("my measurement 2");
+
+  private static final MeasurementUnit MEASUREMENT_UNIT =
+      MeasurementUnit.create(-6, Arrays.asList(BasicUnit.SECONDS));
+
+  private static final String MEASUREMENT_DESCRIPTION = "measurement description";
+
+  private static final MeasurementDescriptor MEASUREMENT_DESCRIPTOR =
+      MeasurementDescriptor.create(MEASUREMENT_NAME, MEASUREMENT_DESCRIPTION, MEASUREMENT_UNIT);
+
+  private static final ViewDescriptor.Name VIEW_NAME = ViewDescriptor.Name.create("my view");
+  private static final ViewDescriptor.Name VIEW_NAME2 = ViewDescriptor.Name.create("my view 2");
+
+  private static final String VIEW_DESCRIPTION = "view description";
+
+  private static final BucketBoundaries BUCKET_BOUNDARIES =
+      BucketBoundaries.create(
+          Arrays.asList(
+              0.0, 0.2, 0.5, 1.0, 2.0, 3.0, 4.0, 5.0, 7.0, 10.0, 15.0, 20.0, 30.0, 40.0, 50.0));
+
+  private static final DistributionAggregationDescriptor DISTRIBUTION_AGGREGATION_DESCRIPTOR =
+      DistributionAggregationDescriptor.create(BUCKET_BOUNDARIES.getBoundaries());
 
-  private static final double TOLERANCE = 1e-5;
-  private static final TagKey tagKey = RpcMeasurementConstants.RPC_CLIENT_METHOD;
-  private static final TagKey wrongTagKey = TagKey.create("Wrong Tag Key");
-  private static final TagKey wrongTagKey2 = TagKey.create("Another wrong Tag Key");
-  private static final TagValue tagValue1 = TagValue.create("some client method");
-  private static final TagValue tagValue2 = TagValue.create("some other client method");
   private final TestClock clock = TestClock.create();
+
   private final StatsManagerImplBase statsManager =
       new StatsManagerImplBase(new SimpleEventQueue(), clock);
-  private final StatsContextImpl oneTag =
-      new StatsContextImpl(statsManager, ImmutableMap.of(tagKey, tagValue1));
-  private final StatsContextImpl anotherTag =
-      new StatsContextImpl(statsManager, ImmutableMap.of(tagKey, tagValue2));
-  private final StatsContextImpl wrongTag =
-      new StatsContextImpl(statsManager, ImmutableMap.of(wrongTagKey, tagValue1));
-  private final StatsContextImpl wrongTag2 =
-      new StatsContextImpl(
-          statsManager, ImmutableMap.of(wrongTagKey, tagValue1, wrongTagKey2, tagValue2));
+
+  private final StatsContextFactoryImpl factory = new StatsContextFactoryImpl(statsManager);
 
   @Test
-  public void testRegisterAndGetView() throws Exception {
-    statsManager.registerView(RpcViewConstants.RPC_CLIENT_ROUNDTRIP_LATENCY_VIEW);
-    View actual = statsManager.getView(RpcViewConstants.RPC_CLIENT_ROUNDTRIP_LATENCY_VIEW);
-    assertThat(actual.getViewDescriptor()).isEqualTo(
-        RpcViewConstants.RPC_CLIENT_ROUNDTRIP_LATENCY_VIEW);
+  public void testRegisterAndGetView() {
+    DistributionViewDescriptor viewDescr =
```

Comment:
I added two helper methods for the two main ways that the tests use DistributionViewDescriptors.  Two tests don't use any attributes of the view, and most others use the name, MeasurementDescriptor, AggregationDescriptor, and tag keys.